src/components/shop/ShopDirective.js

```diff
@@ -0,0 +1,69 @@
+goog.provide('ga_shop_directive');
+goog.require('ga_map_service');
+(function() {
+
+  var module = angular.module('ga_shop_directive', [
+    'ga_map_service'
+  ]);
+
+  module.directive('gaShop', function($window, gaLang, gaLayers) {
+    return {
+      restrict: 'A',
+      templateUrl: function(element, attrs) {
+        return 'components/shop/partials/shop.html';
+      },
+      scope: {
+        map: '=gaShopMap',
+        feature: '=gaShopFeature'
+      },
+      link: function(scope, elt, attrs, controller) {
+
+        // Remove the element if no feature defined
+        if (!scope.feature) {
+          elt.remove();
+          return;
+        }
+
+        var layerBodId = (scope.feature instanceof ol.Feature) ?
+            scope.feature.get('layerId') : scope.feature.layerBodId;
+
+        // Remove the element if no layerBodId associated
+        if (!layerBodId) {
+          elt.remove();
+          return;
+        }
+        var layerConfig = gaLayers.getLayer(layerBodId);
+
+        // Remove the element if no shop config available
+        if (!layerConfig || !layerConfig.shop ||
+            layerConfig.shop.length == 0) {
+          elt.remove();
+          return;
+        }
+
+        scope.orderTypes = layerConfig.shop;
+        scope.orderType = scope.orderTypes[0];
+
+        // {mapsheet,commune,district,canton,rectangle,whole}
+        // Order a mapsheet
+        scope.orderMapsheet = function() {
+          window.open('http://www.toposhop.ch/#/' + gaLang.get() +
```

Comment:
Currently, we don't have any test environment @swisstopo. This is a pending issue,for C2C.
We may use http://shop-shopbgdi.openshift.puzzle.ch but this URL will change later.
Example:
1)
http://shop-shopbgdi.openshift.puzzle.ch/#/de/dispatcher?layers=ch.swisstopo.pixelkarte-lk25.metadata&ch.swisstopo.pixelkarte-lk25.metadata=1011
will add 1011: Beggingen(lk25) to the basket and set the shop language to de.
2)
http://shop-shopbgdi.openshift.puzzle.ch/#/fr/dispatcher?layers=ch.swisstopo.pixelkarte-lk50.metadata&ch.swisstopo.pixelkarte-lk50.metadata=205
will add 205: Schaffhausen (lk50) to the basket and set the shop language to fr.